recipes/libev/all/patches/0001-cmake-for-msvc.patch

```diff
@@ -0,0 +1,254 @@
++#undef HAVE_POLL_H
++
++/* Define to 1 if you have the `port_create' function. */
++#undef HAVE_PORT_CREATE
++
++/* Define to 1 if you have the <port.h> header file. */
++#undef HAVE_PORT_H
++
++/* Define to 1 if you have the `select' function. */
++#undef HAVE_SELECT
++
++/* Define to 1 if you have the `signalfd' function. */
++#undef HAVE_SIGNALFD
++
++/* Define to 1 if you have the <stdint.h> header file. */
++#undef HAVE_STDINT_H
++
++/* Define to 1 if you have the <stdlib.h> header file. */
++#undef HAVE_STDLIB_H
++
++/* Define to 1 if you have the <strings.h> header file. */
++#undef HAVE_STRINGS_H
++
++/* Define to 1 if you have the <string.h> header file. */
++#undef HAVE_STRING_H
++
++/* Define to 1 if you have the <sys/epoll.h> header file. */
++#undef HAVE_SYS_EPOLL_H
++
++/* Define to 1 if you have the <sys/eventfd.h> header file. */
++#undef HAVE_SYS_EVENTFD_H
++
++/* Define to 1 if you have the <sys/event.h> header file. */
++#undef HAVE_SYS_EVENT_H
++
++/* Define to 1 if you have the <sys/inotify.h> header file. */
++#undef HAVE_SYS_INOTIFY_H
++
++/* Define to 1 if you have the <sys/select.h> header file. */
++#undef HAVE_SYS_SELECT_H
++
++/* Define to 1 if you have the <sys/signalfd.h> header file. */
++#undef HAVE_SYS_SIGNALFD_H
++
++/* Define to 1 if you have the <sys/stat.h> header file. */
++#undef HAVE_SYS_STAT_H
++
++/* Define to 1 if you have the <sys/types.h> header file. */
++#undef HAVE_SYS_TYPES_H
++
++/* Define to 1 if you have the <unistd.h> header file. */
++#undef HAVE_UNISTD_H
++
++/* Define to the sub-directory in which libtool stores uninstalled libraries.
++   */
++#undef LT_OBJDIR
++
++/* Name of package */
++#undef PACKAGE
++
++/* Define to the address where bug reports for this package should be sent. */
++#undef PACKAGE_BUGREPORT
++
++/* Define to the full name of this package. */
++#undef PACKAGE_NAME
++
++/* Define to the full name and version of this package. */
++#undef PACKAGE_STRING
++
++/* Define to the one symbol short name of this package. */
++#undef PACKAGE_TARNAME
++
++/* Define to the home page for this package. */
++#undef PACKAGE_URL
++
++/* Define to the version of this package. */
++#undef PACKAGE_VERSION
++
++/* Define to 1 if you have the ANSI C header files. */
++#undef STDC_HEADERS
++
++/* Version number of package */
++#undef VERSION
++
++#define HAVE_SELECT 1
++#define HAVE_SYS_SELECT_H 1
+diff --git a/internal_utils.cmake b/internal_utils.cmake
+new file mode 100644
+index 0000000..f738e79
+--- /dev/null
++++ b/internal_utils.cmake
+@@ -0,0 +1,65 @@
++macro(fix_default_compiler_settings_)
++  if (MSVC)
++    foreach (flag_var
++             CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
++             CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
++      # We prefer more strict warning checking for building Google Test.
++      # Replaces /W3 with /W4 in defaults.
++      string(REPLACE "/W3" "/W4" ${flag_var} "${${flag_var}}")
```

Comment:
🤔 This looks a little out of place. Would you be so kind to explain where this comes from or why it's needed?

(the whole file thats being added I am not sure I understand what problem it's trying to fix)

Reply:
oh, we don't really need this in  cci